src/main/java/uk/ac/cam/cl/dtg/isaac/api/services/EmailService.java

```diff
@@ -51,19 +58,31 @@ public interface HasTitleOrId {
 
     private static final Logger log = LoggerFactory.getLogger(EmailService.class);
 
+    private final PropertiesLoader properties;
     private final EmailManager emailManager;
+    private final MailGunEmailManager mailGunEmailManager;
     private final GroupManager groupManager;
     private final UserAccountManager userManager;
 
     @Inject
-    public EmailService(final EmailManager emailManager, final GroupManager groupManager, final UserAccountManager userManager) {
+    public EmailService(final PropertiesLoader properties, final EmailManager emailManager,
+                        final GroupManager groupManager, final UserAccountManager userManager,
+                        final MailGunEmailManager mailGunEmailManager) {
+        this.properties = properties;
         this.emailManager = emailManager;
         this.groupManager = groupManager;
         this.userManager = userManager;
+        this.mailGunEmailManager = mailGunEmailManager;
     }
 
     private static final DateFormat DATE_FORMAT = new SimpleDateFormat("dd/MM/yy");
-    public void sendAssignmentEmailToGroup(final IAssignmentLike assignment, final HasTitleOrId on, final Map<String, Object> tokenToValueMapping, final String templateName) throws SegueDatabaseException {
+
+    private boolean userInMailGunBetaList(final RegisteredUserDTO user) {
+        String[] ids = properties.getProperty(MAILGUN_EMAILS_BETA_OPT_IN).split(",");
```

Comment:
This should probably check that the property is set, first! (Currently causing a test failure!).

```java
String optInIds = properties.getProperty(MAILGUN_EMAILS_BETA_OPT_IN);
if (Strings.isNullOrEmpty(optInIds)) {
    return false;
}
...
```
(either use the Guava `Strings` helper like that, which will need an import adding, or do `s != null && !s.isEmpty()` yourself).